src/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.OSX.cs

```diff
@@ -59,33 +38,29 @@ public TimeSpan TotalProcessorTime
         {
             get
             {
-                Interop.libproc.proc_threadinfo? info = Interop.libproc.GetThreadInfoById(_processId, (ulong)_threadInfo._threadId);
-                if (info.HasValue)
-                    return new TimeSpan((long)(info.Value.pth_user_time + info.Value.pth_system_time));
-                else
-                    throw new System.ComponentModel.Win32Exception();
+                Interop.libproc.proc_threadinfo info = GetThreadInfo();
+                return new TimeSpan((long)(info.pth_user_time + info.pth_system_time));
             }
         }
 
         /// <summary>
         /// Returns the amount of time the associated thread has spent running code
         /// inside the application (not the operating system core).
         /// </summary>
-        public TimeSpan UserProcessorTime
-        {
-            get
-            {
-                Interop.libproc.proc_threadinfo? info = Interop.libproc.GetThreadInfoById(_processId, (ulong)_threadInfo._threadId);
-                if (info.HasValue)
-                    return new TimeSpan((long)info.Value.pth_user_time);
-                else
-                    throw new System.ComponentModel.Win32Exception();
-            }
-        }
+        public TimeSpan UserProcessorTime => new TimeSpan((long)GetThreadInfo().pth_user_time);
 
         // -----------------------------
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        private Interop.libproc.proc_threadinfo GetThreadInfo()
+        {
+            Interop.libproc.proc_threadinfo? info = Interop.libproc.GetThreadInfoById(_processId, _threadInfo._threadId);
+            if (!info.HasValue)
+            {
+                throw new InvalidOperationException(SR.Format(SR.ThreadExited, Id));
+            }
+            return info.GetValueOrDefault();
```

Comment:
> as we know it has a value?

Right.  ```Value``` does a check to see whether there's a value, whereas ```GetValueOrDefault()``` just returns the field storing the value, without a check.  Since we just did the check ourselves (in order to throw the exception we wanted to throw, we can skip having nullable do it as well.